core/src/main/scala/akka/kafka/ConsumerMessage.scala

```diff
@@ -60,7 +60,7 @@ object ConsumerMessage {
     def commitJavadsl(): CompletionStage[Done]
 
     @InternalApi
-    private[kafka] def commitInternal(): Future[Done]
+    private[kafka] def commitInternal(flush: Boolean = false): Future[Done]
```

Comment:
Instead of a the parameter here, try and add a method `private[kafka] def commitEmergency(): Future[Done]` to `CommittableOffsetBatch`. It makes this less intrusive.

Reply:
I actually like the semantics of `flush` more (also makes the usage compact and descriptive) and with default arguments it's still not as intrusive. 
I've added the method, this required some minor refactoring. Please have a look if this is ok.

Reply:
Will tellCommit() would also require en `tellCommitEmergency()` method?

Reply:
@ennru please also see my other remark. Maybe we should just have one `tellCommitEmergency()` (or `commitEmergency()` for simplicity sake)?

Reply:
> @ennru please also see my other remark. Maybe we should just have one `tellCommitEmergency()` (or `commitEmergency()` for simplicity sake)?

On the other hand, `akka.kafka.internal.CommittingProducerSinkStageLogic#commit` uses callbacks, waiting for your feedback on this. For now leaving as it is.